cmd/spire-server/cli/agent/purge.go

```diff
@@ -0,0 +1,156 @@
+package agent
+
+import (
+	"flag"
+	"fmt"
+	"time"
+
+	"github.com/mitchellh/cli"
+	"github.com/spiffe/go-spiffe/v2/spiffeid"
+	agentv1 "github.com/spiffe/spire-api-sdk/proto/spire/api/server/agent/v1"
+	"github.com/spiffe/spire-api-sdk/proto/spire/api/types"
+	"github.com/spiffe/spire/cmd/spire-server/util"
+	commoncli "github.com/spiffe/spire/pkg/common/cli"
+	"github.com/spiffe/spire/pkg/common/cliprinter"
+	"github.com/spiffe/spire/pkg/common/idutil"
+	"golang.org/x/net/context"
+	"google.golang.org/protobuf/types/known/wrapperspb"
+)
+
+type purgeCommand struct {
+	env        *commoncli.Env
+	expiredFor time.Duration
+	dryRun     bool
+	printer    cliprinter.Printer
+}
+
+func NewPurgeCommand() cli.Command {
+	return NewPurgeCommandWithEnv(commoncli.DefaultEnv)
+}
+
+func NewPurgeCommandWithEnv(env *commoncli.Env) cli.Command {
+	return util.AdaptCommand(env, &purgeCommand{env: env})
+}
+
+func (*purgeCommand) Name() string {
+	return "agent purge"
+}
+
+func (*purgeCommand) Synopsis() string {
+	return "Delete expired agents that attested using a non-TOFU security model based on a given time"
+}
+
+func (c *purgeCommand) Run(ctx context.Context, _ *commoncli.Env, serverClient util.ServerClient) (err error) {
+	agentClient := serverClient.NewAgentClient()
+	resp, err := agentClient.ListAgents(ctx, &agentv1.ListAgentsRequest{
+		Filter:     &agentv1.ListAgentsRequest_Filter{ByCanReattest: wrapperspb.Bool(true)},
+		OutputMask: &types.AgentMask{X509SvidExpiresAt: true},
+	})
+	if err != nil {
+		return fmt.Errorf("failed to list agents: %w", err)
+	}
+
+	agents := resp.GetAgents()
+	expiredAgents := &expiredAgents{Agents: []*expiredAgent{}}
+
+	for _, agent := range agents {
+		id, err := idutil.IDFromProto(agent.Id)
+		if err != nil {
+			return err
+		}
+
+		expirationTime := time.Unix(agent.X509SvidExpiresAt, 0)
+
+		if time.Since(expirationTime) > c.expiredFor {
+			result := &expiredAgent{AgentID: id}
+
+			if !c.dryRun {
+				if _, err := agentClient.DeleteAgent(ctx, &agentv1.DeleteAgentRequest{Id: agent.Id}); err != nil {
+					result.Error = err.Error()
+				} else {
+					result.Deleted = true
+				}
+			}
+			expiredAgents.Agents = append(expiredAgents.Agents, result)
+		}
+	}
+
+	return c.printer.PrintStruct(expiredAgents)
+}
+
+func (c *purgeCommand) AppendFlags(fs *flag.FlagSet) {
+	fs.DurationVar(&c.expiredFor, "expiredFor", 24*time.Hour, "Specifies the time since the agent's SVID has expired, used for filtering agents to purge.")
```

Comment:
```suggestion
	fs.DurationVar(&c.expiredFor, "expiredFor", 24*time.Hour, "Amount of time that has passed since the agent's SVID has expired. It is used to determine which agents to purge.")
```

Reply:
06c00a4